lib/src/bridge/scanning_mixin.dart

```diff
@@ -40,8 +42,8 @@ mixin ScanningMixin on FlutterBLE {
     );
 
     streamController
-        .addStream(_scanEvents, cancelOnError: true)
```

Comment:
@mikolak the problem lies here. If I leave it _scanEvents then the compiler is just not clever enough to know that _scanEvents is actually autoUnwrappable, and I'll have to write: `.addStream(_scanEvents!, cancelOnError: true)`
It's not that force unwrap operator is ugly, it's discouraged to use for a reason:
- Imagine, if the code above changes any time in the future, changing the behaviour of the code in way that `_scanEvents` is no longer guaranteed to be `nonnull`.
Nothing is gonna warn you about the `!` force unwrap now being semantically incorrect, and crashing the software on edge cases.
- Whereas if you maintain a helper local var instead /more verbosely I know/, then this behaviour change on 43 will break the compilation of the code since the compiler will see that `scanEvents` local var can no longer be proven `nonnull` and thus it will be `nullable` and `addStream()` don't accept no nullable value by its signature, so it will not compile, which is a good thing. It can safeguard you from future mistakes even, warning you to add an extra null check or something.

So yeah, it may seem to be an unnecessary complication with the extra var and all, but it really for semanticly correct nullsafe coding. 
Ofc, I'm also happy with anything that achieves the same, correct me if I'm wrong but I don't see you suggested change achieving the same safety concern.
Or b.) no fucks given and I should just make it less verbose sacrificing nullsafety.
Up to you ;)